hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -248,7 +250,9 @@ public void deleteByTenant(final String tenant) {
         ObjectListing objects = amazonS3.listObjects(s3Properties.getBucketName(), folder + "/");
         do {
             for (final S3ObjectSummary objectSummary : objects.getObjectSummaries()) {
-                amazonS3.deleteObject(s3Properties.getBucketName(), objectSummary.getKey());
+                final S3Object s3Object = amazonS3.getObject(s3Properties.getBucketName(), objectSummary.getKey());
```

Comment:
I don't think it's necessary to retrieve the `S3Object` and close it immediately - the `S3ObjectSummary` does not need to be closed and the `S3Object` is not needed. Why you would retrieve it here to immediately close it again?

Reply:
Perhaps I misunderstand something. My understanding for the precondition of this loop is that there exists some number of S3ObjectSummary objects, each of which contains a key used to find an associated S3Object in a bucket. It's that associated S3Object, not the S3ObjectSummary, that the code is trying to close. However, if the lifetime of the S3Object is longer than the S3ObjectSummary, then the code (both new and existing) is not correct.

If I understand the conditions correctly, I will change the code to:
```
closeS3Object(amazonS3.getObject(s3Properties.getBucketName(), objectSummary.getKey());
```

Reply:
@michahirsch When you have an opportunity, I'd appreciate feedback. Thanks.

Reply:
@michahirsch Feedback, please? Thanks.